aea/cli/ipfs_hash.py

```diff
@@ -0,0 +1,491 @@
+
+
+def _get_all_packages(packages_dir: Path) -> List[Tuple[PackageType, Path]]:
+    """Get all the hashable package of the repository."""
+
+    return list(
+        map(
+            package_type_and_path,
+            filter(operator.methodcaller("is_dir"), packages_dir.glob("*/*/*/")),
+        )
+    )
+
+
+def sort_configuration_file(config: PackageConfiguration) -> None:
+    """Sort the order of the fields in the configuration files."""
+    # load config file to get ignore patterns, dump again immediately to impose ordering
+    if config.directory is None:
+        raise ValueError("config.directory cannot be None.")
+
+    configuration_filepath = config.directory / config.default_configuration_filename
+    if config.package_type == PackageType.AGENT:
+        json_data = config.ordered_json
+        component_configurations = json_data.pop("component_configurations")
+        yaml_dump_all(
+            [json_data] + component_configurations, configuration_filepath.open("w")
+        )
+    else:
+        yaml_dump(config.ordered_json, configuration_filepath.open("w"))
+
+
+def hash_package(
+    configuration: PackageConfiguration, package_type: PackageType, wrap: bool = True
+) -> Tuple[str, str]:
+    """
+    Hashes a package and its components.
+
+    :param configuration: the package configuration.
+    :param package_type: the package type.
+    :param wrap: Whether to use the wrapper node or not.
+    :return: the identifier of the hash (e.g. 'fetchai/protocols/default')
+           | and the hash of the whole package.
+    """
+    # hash again to get outer hash (this time all files)
+    # we still need to ignore some files
+    #      use ignore patterns somehow
+    # ignore_patterns = configuration.fingerprint_ignore_patterns # noqa: E800
+    if configuration.directory is None:
+        raise ValueError("configuration.directory cannot be None.")
+
+    key = os.path.join(
+        configuration.author, package_type.to_plural(), configuration.directory.name,
+    )
+    package_hash = HASH_TOOL.hash_directory(str(configuration.directory), wrap=wrap)
+    return key, package_hash
+
+
+def to_csv(package_hashes: Dict[str, str], path: Path) -> None:
+    """Outputs a dictionary to CSV."""
+    try:
+        ordered = collections.OrderedDict(sorted(package_hashes.items()))
+        with open(path, "w") as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerows(ordered.items())
+    except IOError:
+        click.echo("I/O error")
+
+
+def from_csv(path: Path) -> Dict[str, str]:
+    """Load a CSV into a dictionary."""
+    result = collections.OrderedDict({})  # type: Dict[str, str]
+    with open(path, "r") as csv_file:
+        reader = csv.reader(csv_file)
+        for row in reader:
+            if len(row) != 2:
+                raise ValueError("Length of the row should be 2.")
+
+            key, value = row
+            result[key] = value
+    return result
+
+
+def load_configuration(
```

Comment:
This must exist in a similar form already elsewhere in the framework; find it and reuse

Reply:
@angrybayblade